spec/ddtrace/workers/trace_writer_spec.rb

```diff
@@ -602,14 +602,60 @@
     end
   end
 
+  describe '#thread_cpu_time_diff' do
+    def subject
+      writer.send(:thread_cpu_time_diff)
+    end
+
+    it 'calculates difference since last measurement' do
+      expect(Datadog::Utils::Time).to receive(:get_thread_cpu_time).and_return(1, 10)
+
+      is_expected.to eq(1)
+      is_expected.to eq(9)
+    end
+  end
+
   describe 'integration tests' do
-    let(:options) { { transport: transport, fork_policy: fork_policy } }
+    let(:options) { { transport: transport } }
     let(:transport) { Datadog::Transport::HTTP.default { |t| t.adapter :test, output } }
     let(:output) { [] }
 
+    describe 'health metrics' do
+      include_context 'health metrics'
+
+      let(:trace) { get_test_traces(1).first }
+      let(:writer_cpu_time) { double }
+
+      before do
+        allow(writer).to receive(:thread_cpu_time_diff).and_return(writer_cpu_time)
+      end
+
+      after do
+        if Datadog::Utils::Time::THREAD_CPU_TIME_SUPPORTED
+          # TODO: There could be leaky background worker threads (created by other tests) that
+          # TODO: might be sending metrics to `Datadog.health_metrics`.
+          # TODO: We do our best here to assert that metric we expect was recorded.
+          # TODO: Ideally we should clean up threads created on every unit test.
```

Comment:
We create a lot of worker threads throughout our test suite that are not properly closed. During a run of `rake spec:main`, we end up with dozens of threads by the last test.

This is something we should thing about how to address across the board.

Reply:
agreed. is there something we can do here to patch/monitor `Datadog.health_metrics` just for this single thread?

Reply:
I don't know of a way to do that, given the object `Datadog.health_metrics` is global. I tried stubbing the constant `::Datadog` and the object `Datadog.health_metrics`, but all threads have access to it.

The only way I can think is by introducing `Datadog.health_metrics` as an explicit dependency to our two Worker implementations, which is valid but also not ideal for cross-cutting concerns like a health metric logger.